wallet-new/src/Cardano/Wallet/API/V1/Types.hs

```diff
@@ -1767,6 +1770,31 @@ instance BuildableSafeGen NodeInfo where
         nfoLocalTimeInformation
         (Map.toList nfoSubscriptionStatus)
 
+-- | A shielded redemption code.
+newtype ShieldedRedemptionCode = ShieldedRedemptionCode
+    { unShieldedRedemptionCode :: Text
+    } deriving (Eq, Show, Generic)
+
+-- | The request body for redeeming some Ada.
+data Redemption = Redemption
+    { redemptionWalletId       :: WalletId
+    -- ^ The 'WalletId' that the redemption will go toward.
+    , redemptionAccountIndex   :: AccountIndex
+    -- ^ The index of the account in the wallet that the redemption applies
+    -- to.
```

Comment:
These (`redemptionWalletId` & `redemptionAccountIndex`) would be moved to function arguments if we indeed proceed with a

`api/v1/wallets/{walletId}/accounts/{accountId}/certificates`

endpoint :+1:

Reply:
Good call, I have done this :)